### ISSUE TITLE
identifier_name issue didn't show warning 

### ISSUE THREAD

**Issue by ilikerb0407** [realm/SwiftLint#4767]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

 In an exception to the above, variable names may start with a capital letter when they are **declared static and immutable.**
Above is the sentence on SwiftLint Rules, I found out that when I declared  `static var AbCvar = 0`, and Xcode didn't show the warning because I realize the rule mention static and immutable. What is the reason ?

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.50.3
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:

```yml
excluded:
  - Pods
disabled_rules:
  - trailing_whitespace
identifier_name:
  excluded: # excluded via string array
    - id
    - URL
    - url
    - app
file_length:
  warning: 500
  error: 800
type_body_length:
  - 500 # warning
  - 700 # error
```

**Comment by SimplyDanny** [comment]
The trigger for this exception was #541 implemented in #1444. However, I cannot find the reason for it. The style guides I've scrolled through don't mention special rules for immutable constants.

The rule has been like that for a long time now. That's why I'd rather avoid to change its behavior. The exception for static variables is there but can be disabled by a configuration option. So if you don't like it, just deactivate this exception. 😉 

Yet, you are right that there is an inconsistency in the rule's description and its behavior in that the exception is only valid for **immutable** static variables while the implementation checks all static variables no matter if they are constants or not.

Again, to avoid bahavioral changes, we should go and fix the description only.

**Comment by mildm8nnered** [comment]
#4784 should resolve this.

Somewhat tangential to the original complaint, it does seem a bit weird that

```
class Foo {
    static let Bar = 0
}
```

(or it's `var` equivalent), does not trigger this rule, but that a top level (i.e. global) static declaration will

e.g. `static let Bar = 0` or `static var Bar = 0`

**Comment by JaviSoto** [comment]
> The exception for static variables is there but can be disabled by a configuration option

Which option is this? Am I missing something? I don't see it in https://realm.github.io/SwiftLint/identifier_name.html

**Comment by SimplyDanny** [comment]
> > The exception for static variables is there but can be disabled by a configuration option
> 
> Which option is this? Am I missing something? I don't see it in https://realm.github.io/SwiftLint/identifier_name.html

Well, the case check can be disabled using `validates_start_with_lowercase` for all variables, so that it doesn't matter anymore. It's not that it can be enabled for static variables. Sorry, if my previous statement was a bit confusing.
